clusterloader2/pkg/prometheus/gce_windows_util.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prometheus
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/config"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework/client"
+)
+
+// gceNode is a struct storing gce node info
+type gceNode struct {
+	projectID string
+	zone      string
+	nodeName  string
+}
+
+func setUpWindowsNodeAndTemplate(k8sClient kubernetes.Interface, mapping map[string]interface{}) error {
+	// Get the gce windows node info
+	windowsNode, err := getGceWindowsNodeFromKubernetesService(k8sClient)
+	if err != nil {
+		return err
+	}
+	// Install the wmi exporter onto windows node
+	if err := installWmiExporter(k8sClient, windowsNode, mapping); err != nil {
+		return err
+	}
+	mapping["WINDOWS_NODE_NAME"] = windowsNode.nodeName
+	mapping["PROMETHEUS_SCRAPE_WINDOWS_NODES"] = true
+	return nil
+}
+
+func enableScrapingWindowsNode(mapping map[string]interface{}, clusterLoaderConfig *config.ClusterLoaderConfig) bool {
```

Comment:
I'd prefer changing the name, `enableScrapingWindowsNode` suggests it's actually enabling something while it only checks whether something should be enabled or not. What about `shouldEnableScrapingWindowsNodes` or `isWindowsNodeScrapingEnabled` ?

Reply:
Agree, changed to isWindowsNodeScrapingEnabled